viewing_party/party.py

```diff
@@ -1,23 +1,156 @@
+            genre_freq[movie_genre] = 1
+        else:
+            genre_freq[movie_genre] += 1
+        
+    #initially I used this code to find the most frequent genre using the values of the dictionary (which represents the freqnecy), and figuring out the key. But commented it out bc I found a one liner online to make it shorter.
+    #list_of_popular_genres = []
+    #max_frequency = 0
+    #for genre in genre_freq:
+    #    if genre_freq[genre] > max_frequency:
+    #        max_frequency = genre_freq[genre]
+    #        list_of_popular_genres.append(genre)
+    #return list_of_popular_genres[-1] 
+
+    return max(genre_freq, key=genre_freq.get)
+    
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):
+    list_of_titles_watched_by_friends = []
+    for friend in user_data["friends"]:
+          for movie in friend["watched"]:
+            list_of_titles_watched_by_friends.append(movie["title"])
+
+    list_of_titles_watched_by_user = []
+    for movie in user_data["watched"]:
+        list_of_titles_watched_by_user.append(movie["title"])
+    
+    #using sets to find the difference!
+    set_of_friends_movies = set(list_of_titles_watched_by_friends)
+    set_of_users_movies = set(list_of_titles_watched_by_user)
+    set_of_movies_watched_by_user_only = set_of_users_movies - set_of_friends_movies
+
+    list_of_unique_movies_watched = []
+
+    #for each movie in the user's watched list, if the title appears in the 
+    #set of unique movies, then we add it to the list we intend to return
+    for movie in user_data["watched"]:
+        if movie["title"] in set_of_movies_watched_by_user_only:
+            list_of_unique_movies_watched.append(movie)
+
+    return list_of_unique_movies_watched
+
+def get_friends_unique_watched(user_data):
+    list_of_titles_watched_by_friends = []
+    for friend in user_data["friends"]:
+          for movie in friend["watched"]:
+            list_of_titles_watched_by_friends.append(movie["title"])
+
+    list_of_titles_watched_by_user = []
+    for movie in user_data["watched"]:
+        list_of_titles_watched_by_user.append(movie["title"])
+    
+    #using sets to find the difference!
+    set_of_friends_movies = set(list_of_titles_watched_by_friends)
+    set_of_users_movies = set(list_of_titles_watched_by_user)
+    set_of_movies_watched_by_friends_only = set_of_friends_movies - set_of_users_movies 
+
+    list_of_unique_movies_watched = []
+
+    #for each friend, and each movie in their list, if the title appears in the 
+    #set of unique movies, then we add it to the list we intend to return
+    for friend in user_data["friends"]:
+          for movie in friend["watched"]:
+            if movie["title"] in set_of_movies_watched_by_friends_only and movie not in list_of_unique_movies_watched:
+                list_of_unique_movies_watched.append(movie)

```

Comment:
hmm where have we seen this code before? A lot of it is very similar to the code in `get_unique_watched`. I think we could turn these blocks of code into helper functions! That way we don't have to repeat the same lines again.